great_tables/_locations.py

```diff
@@ -869,14 +917,21 @@ def _(loc: LocStub, data: GTData) -> set[int]:
 @resolve.register
 def _(loc: LocBody, data: GTData) -> list[CellPos]:
     cols = resolve_cols_i(data=data, expr=loc.columns)
-    rows = resolve_rows_i(data=data, expr=loc.rows)
-
-    # TODO: dplyr arranges by `Var1`, and does distinct (since you can tidyselect the same
-    # thing multiple times
-    cell_pos = [
-        CellPos(col[1], row[1], colname=col[0]) for col, row in itertools.product(cols, rows)
-    ]
+    if (loc.columns is not None or loc.rows is not None) and loc.mask is not None:
+        raise ValueError(
+            "Cannot specify the `mask` argument along with `columns` or `rows` in `loc.body()`."
+        )
 
+    if loc.mask is None:
+        rows = resolve_rows_i(data=data, expr=loc.rows)
```

Comment:
Can we move `cols = resolve_cols_i(...)` from line 920 to this if block? That way the guard that errors if columns and mask are both set triggers right away, and the cols piece only runs if needed.

(I could be missing an important reason for it to be at the top though)